### DISCUSSION TITLE
How to disable/increase memory size allocation limit?

### DISCUSSION THREAD

**Discussion by GallantArc302** [gildor2/UEViewer#250] [Q&A] [answered]
So I'm trying to extract models from FNaF Security Breach, but it keeps saying "Memory: bad allocation size" when I try to open or extract some of the models. I looked around a bit more and found [this](https://www.gildor.org/smf/index.php?topic=3333.0) which mentions disabling a check for "bad memory block size" but they don't mention where or how.

**Comment by gildor2** [comment, marked as answer]
That message on the forum was made by the guy, who doesn't know what he says. Just ignore it. Supporting for some game in UModel is NOT as easy as "disabling some error in code". If Windows crashes with Blue Screen - just DISABLE this "Blue Screen", right?

**Reply by GallantArc302** [reply]
Aren't you the one who replied to the post saying to "disable check for "bad memory block size"" though? To me it looks like the reason it crashes is because the size of the file goes over the limit you coded (512MB). Increasing that limit to around 2048MB should allow it to open bigger files, right?

**Reply by gildor2** [reply]
Well. I've re-read the thread. Firstly, there was a discussion about working with 16k x 16k textures, which are really huge, and there's a real limit in UModel - to not let to allocate a memory block (not "file size", but a "single contiguous memory block size") larger than 512k, because it's untypical.

Then a some guy "joined" discussion with "how to disable limit" message. It's not his first post there - he posted things before. He thinks that if he'll disable this limit, then he'll open Fortnite or so on with no problems. In his opinion, I'm stupid, added artificial limitations to my own code, and he's cleaver (but can't find how to do trivial things). So - again, just ignore that user. In his earlier post, UModel crashed with allocating NEGATIVE amount of memory. He wants UModel to let doing this. Then he'll enter the 5th dimension,

**Reply by GallantArc302** [reply]
Ok I understand now, thanks. Hope you figure out how to get Security Breach working

**Reply by gamerbross** [reply]
select ue4.25 plus
![image](https://user-images.githubusercontent.com/55158797/149411598-49800bec-7a6e-45f3-912e-04514ce451ab.png)
![image](https://user-images.githubusercontent.com/55158797/149411765-cde09d43-8aa5-4304-9433-8022f4ada446.png)
![image](https://user-images.githubusercontent.com/55158797/149411808-7b248721-3c14-42ea-9049-70583af45b91.png)
![image](https://user-images.githubusercontent.com/55158797/149411818-a320fab2-29ce-411d-8cb2-603da94ab7a7.png)
(only works when bytes are negative)

**Reply by Creepaster** [reply]
i always have a error...

the message appear, and unmodel instantly crash,

******** /Game/Maps/World/MAP_NightLighting_BuiltData.uasset ********

Serialize bulk data: current position FFFFFFFF8005E000, expected 8005E000
ERROR: FPakFile::Seek64(0x8005E000) - not implemented
FByteBulkData::SerializeData <- FByteBulkData::SerializeData(UObject*) <- UTexture2D::LoadBulkTexture: File=none Mip=1 <- UTexture2D::GetTextureData: LQ_Lightmap_0_231 <- CTextureExportWorker::Setup <- ExportTexture <- ExportMaterial: LightMapTexture2D'LQ_Lightmap_0_231' <- ExportObject: LightMapTexture2D'LQ_Lightmap_0_231' <- ExportObjects <- ExportPackages <- CUmodelApp::ShowPackageUI <- Main: umodel_build=1585


so i turn off the "lightmap texture" and..
******** Loading object StaticMesh4'MOD_StaticFoxy.MOD_StaticFoxy' ********

ERROR: Memory: bad allocation size 565063704 bytes
appMalloc: size=565063704 (total=194 Mbytes) <- FArray::Empty: 1097286145 x 24 <- TArray::Serialize: 0/0 <- FStaticMeshLODModel4<< <- TArray::Serialize: 2/4 <- UStaticMesh4::Serialize <- LoadObject: StaticMesh4'MOD_StaticFoxy.MOD_StaticFoxy', pos=23A75A, ver=518/0 (unversioned), game=ue4.25+ <- UObject::EndLoad <- LoadWholePackage: /Game/Model_Assets/Chars/MOD_StaticFoxy.uasset <- ExportPackages <- CUmodelApp::ShowPackageUI <- Main: umodel_build=1585

i use the 4.25 PLUS version...

**Reply by ghost** [reply]
Something for UE 4.26?

**Reply by Bluscream** [reply]
Same here:

Memory: bad allocation size 1774385215 bytes
appMalloc: size=1774385215 (total=0 Mbytes) <- FArray::Empty: 1774385215 x 1 <- FString<< <- FPackageFileSummary::Serialize3 <- FPackageFileSummary<<: Ver=1809/0 <- UnPackage::UnPackage: CombatGame/CookedPC/025e2e055816ded40.upk, ver=1809/0, game=ue3 <- UnPackage::LoadPackage(info): CombatGame/CookedPC/025e2e055816ded40.upk <- ScanContent <- UIMenuItem::HandleCommand <- UIMenu::Popup <- UIBaseDialog::WndProc <- UIBaseDialog::CustomMessageLoop <- UIBaseDialog::ShowDialog: modal=1, title="Choose a package to open" <- CUmodelApp::ShowPackageUI <- Main: umodel_build=1590
